shaky/src/main/java/com/linkedin/android/shaky/Utils.java

```diff
@@ -130,4 +140,117 @@ static Uri getProviderUri(@NonNull Context context, @NonNull Uri uri) {
         File file = new File(uri.getPath());
         return getProviderUri(context, file);
     }
+
+    /**
+     * @param uri The uri to evaluate.
+     * @return Whether or not the given uri points to resource on the device. This check is basic, so it does not
+     * guarantee the resource actually exists.
+     */
+    static boolean isLocalUri(@Nullable Uri uri) {
+        if (uri != null) {
+            String scheme = uri.getScheme();
+            return ContentResolver.SCHEME_CONTENT.equalsIgnoreCase(scheme)
+                || ContentResolver.SCHEME_ANDROID_RESOURCE.equalsIgnoreCase(scheme)
+                || ContentResolver.SCHEME_FILE.equalsIgnoreCase(scheme);
+        }
+        return false;
+    }
+
+    /**
+     * Tries to determine the current mime type of the pending attachment by reading the file at it's uri. There's
+     * fallback logic included.
+     *
+     * @param context A context used to read the file at the attachment's uri.
+     * @param uri     Media uri
+     */
+    @Nullable
+    static String getMimeType(@NonNull Context context, @NonNull Uri uri) {
+        if (!isLocalUri(uri)) {
+            return null;
+        }
+        String newMediaType = null;
+
+        // Try to decode the bitmap to get its mime type. In some cases on Android M, the decoder may not return a
+        // mime type. Fallback logic kicks in afterwards.
+        InputStream inputStream = null;
+        try {
+            inputStream = context.getContentResolver().openInputStream(uri);
+            BitmapFactory.Options options = new BitmapFactory.Options();
+            options.inJustDecodeBounds = true;
+            BitmapFactory.decodeStream(inputStream, null, options);
+
+            if (options.outMimeType != null) {
+                newMediaType = options.outMimeType;
+            }
+        } catch (FileNotFoundException e) {
+            Log.e(TAG, "Error getting mediaType for : " + uri.toString(), e);
+        } finally {
+            if (inputStream != null) {
+                try {
+                    inputStream.close();
+                } catch (IOException e) {
+                    // ignore
+                }
+            }
+        }
+
+        // Fallback to the content resolver if the bitmap decoding hasn't worked.
+        if (newMediaType == null) {
+            newMediaType = context.getContentResolver().getType(uri);
+        }
+
+        // Fallback to the file extension if everything else fails.
+        if (newMediaType == null) {
+            String extension = MimeTypeMap.getFileExtensionFromUrl(uri.toString());
+            if (extension != null) {
+                newMediaType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension);
+            }
+        }
+
+        return newMediaType;
+    }
+
+    /**
+     * Make the file uri persistent
+     */
+    static void persistFilePermissions(@NonNull Context context, @NonNull Uri uri, @Nullable Intent data) {
```

Comment:
I'm not sure i understand why we need this, can you explain a bit?

Reply:
I think we don't need this for external shared documents. This is only for if we want to share the internal files.
I will remove it

Reply:
Not sure what you mean by "internal files" - this only supports files on the local disk right?